android/src/main/java/com/doublesymmetry/trackplayer/module/MusicModule.kt

```diff
@@ -27,50 +26,53 @@ import javax.annotation.Nonnull
 
 /**
  * @author Milen Pivchev @mpivchev
- *
  */
-class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext), ServiceConnection, LifecycleEventsListener {
-    private var binder: MusicService.MusicBinder? = null
+class MusicModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext), ServiceConnection, ActivityLifecycleCallbacksAdapter {
     private var eventHandler: MusicEvents? = null
     private var playerOptions: Bundle? = null
     private var isServiceBound = false
     private var playerSetUpPromise: Promise? = null
 
     private lateinit var musicService: MusicService
 
+    val context = reactContext
+
     @Nonnull
     override fun getName(): String {
         return "TrackPlayerModule"
     }
 
     override fun initialize() {
-        val context: ReactContext = reactApplicationContext
-        context.addLifecycleEventListener(this)
+        currentActivity?.application?.registerActivityLifecycleCallbacks(this)
         Logger.addLogAdapter(AndroidLogAdapter())
     }
 
-    /**
-     * Called when the React context is destroyed or reloaded.
-     */
-    override fun onCatalystInstanceDestroy() {
-        if (!isServiceBound) return
+    override fun onActivityStarted(activity: Activity) {
+        // Service MUST be rebound during activity onStart, if not already bound
+        if (isServiceBound) return
 
-        musicService.destroyIfAllowed(true)
-        unbindFromService()
+        Intent(context, MusicService::class.java).also { intent ->
+            context.bindService(intent, this, Context.BIND_AUTO_CREATE)
+        }
     }
 
-    /**
-     * Called when host activity receives destroy event (e.g. {@link Activity#onDestroy}. Only called
-     * for the last React activity to be destroyed.
-     */
-    override fun onHostDestroy() {
-        destroyIfAllowed()
+    override fun onActivityStopped(activity: Activity) {
+        // Service MUST be unbound during activity onStop
+        unbindService()
```

Comment:
Unbinding seems to kill the service automatically. Usually this is prevented by calling `ContextCompat.startForegroundService` to manually start the service, but it seems like it still doesn't work.